### DISCUSSION TITLE
Docs: Addressing "API Routes Response Size Limited to 4MB" Error in Next.js

### DISCUSSION THREAD

**Discussion by lucajung** [vercel/next.js#76523] [Help] [answered]
### What is the improvement or update you wish to see?

"Addressing "API Routes Response Size Limited to 4MB" Error in Next.js" does not seem to apply to App Router, although App Router has been selected.

### Is there any context that might help us understand?

Is there another solution for the App Router? Then please add it to this document.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/messages/api-routes-response-size-limit

**Comment by ozd3v** [comment]
what is the solution or workaround for app router ?

**Comment by adilapapaya** [comment]
Bumping this up as there still doesn't seem to be a solution even though app router is the new standard.

When I try to build with this in my `route.js` file, 
```js
export const config = {
  api: {
    responseLimit: '5mb',
  },
}
```
I get this error:
```
Error: Page config in /next-example/src/app/api/load/route.js is deprecated. Replace `export const config=…` with the following:
Visit https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config for more information.
```

However, https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config  doesn't any any options for setting the api  response limit.

What is the solution / workaround for app router?

**Comment by dacxjo** [comment]
Is there any update on this?

_Edit by maintainer bot: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_

**Comment by dacxjo** [comment]
Seems that in app router you have to do:

```typescript
const nextConfig = {
  experimental: {
    serverActions: {
      bodySizeLimit: '15mb'
    }
  }
}
``` 

in the next.config.js file

**Comment by ozd3v** [comment]
is it solved???

**Comment by jurgen-siegel** [comment]
looking for a solution as well

**Comment by raminrez** [comment]
just use` .js `or `.cjs` as  `next.config` extention .it will fix the error!!! (next.config.js, next.config.cjs)
@lucajung @adilapapaya

**Comment by samcx** [comment, marked as answer]
@lucajung A response size limit does not apply to Route Handlers because the response is streamed, unlike API Routes.

We should clarify this in the documentation 👀
